### PR TITLE
Update win-crowdstrike-fix-bo otloop-v2.ps1 so that registry load/unload is independent with the crowdstrike file check

### PR DESCRIPTION
**Pull request by zmyzheng**
1. Check if corrupt CrowdStrike files exist in each drive letter. Remove any corrupt CrowdStrike files
2. Check if registry config files exist in each non system drive letter. If registry config files exist in the non system drive letter, load it to the registry hive and then unload it.
3. Delete regtrans-ms and txr.blf files under config\TxR for Windows Server 2016 or newer version

image

### CODE REVIEW

File: src/windows/win-crowdstrike-fix-bootloop-v2.ps1 [hunk 1]
```diff
@@ -1,11 +1,124 @@
...
+    Log-Info "System drive is: $systemDrive"
+
+    $registryConfigFileFound = $false
+    forEach ( $partition in $partitionlist )
+    {
+        $driveLetter = $partition.DriveLetter
+        if ($driveLetter) { # Skip partitions without drive letter
+            $driveLetter = ($driveLetter + ":")
+            Log-Info "Check Drive letter: $driveLetter"
+            if ($driveLetter -ne $systemDrive) { # Skip OS disk
```
Comment (maheshnemichand) [inline src/windows/win-crowdstrike-fix-bootloop-v2.ps1:62]: can we make this case insensitive check?

File: src/windows/win-crowdstrike-fix-bootloop-v2.ps1 [hunk 2]
```diff
@@ -1,11 +1,124 @@
...
+                    "HKLM\temp_system_hive_$guidSuffix" = "$driveLetter\windows\system32\config\system"
+                    "HKLM\temp_software_hive_$guidSuffix" = "$driveLetter\windows\system32\config\software"
+                }
+
+                foreach ($regKey in $regKeyToFile.Keys)
+                {
+                    $regFile = $regKeyToFile[$regKey]
+                    if (Test-Path -Path $regFile) {
+                        Log-Info "Found registry config file at $regFile."
+                        $configExixst = $true
```
Comment (maheshnemichand) [inline src/windows/win-crowdstrike-fix-bootloop-v2.ps1:78]: typo in Exist

File: src/windows/win-crowdstrike-fix-bootloop-v2.ps1 [hunk 3]
```diff
@@ -1,11 +1,124 @@
...
+
+                        Log-Info "Loading registry hive $regKey from $regFile..."
+                        $result = reg load $regKey $regFile 2>&1
+                        if ($LASTEXITCODE -ne 0) {
+                            Log-Error "Load registry hive $regKey from $regFile failed with error: $result"
```
Comment (maheshnemichand) [inline src/windows/win-crowdstrike-fix-bootloop-v2.ps1:83]: Should we log LASTEXITCODE as well?

File: src/windows/win-crowdstrike-fix-bootloop-v2.ps1 (hunk 1 above)
Comment (zmyzheng) [reply src/windows/win-crowdstrike-fix-bootloop-v2.ps1:62]: The "Not Equal" comparison here is not case sensitive:
image

We should not do case sensitive comparation as "C:" drive and "c:" drive is the same drive in Windows.

File: src/windows/win-crowdstrike-fix-bootloop-v2.ps1 (hunk 2 above)
Comment (zmyzheng) [reply src/windows/win-crowdstrike-fix-bootloop-v2.ps1]: Thanks for catching this. Fixed

File: src/windows/win-crowdstrike-fix-bootloop-v2.ps1 (hunk 3 above)
Comment (zmyzheng) [reply src/windows/win-crowdstrike-fix-bootloop-v2.ps1]: Sounds good. I just added $LASTEXITCODE into the log-Error

**Review by EdwinBernal1 — APPROVED** [review]

File: src/windows/win-crowdstrike-fix-bootloop-v2.ps1 (hunk 1 above)
Comment (maheshnemichand) [reply src/windows/win-crowdstrike-fix-bootloop-v2.ps1:62]: I agree and was saying case insensitive, which looks like -ne does by default. So, that's good.

**Review by maheshnemichand — APPROVED** [review]

**Pull request merged**